Library/Homebrew/formula_cellar_checks.rb

```diff
@@ -311,6 +345,18 @@ def audit_installed
   def relative_glob(dir, pattern)
     File.directory?(dir) ? Dir.chdir(dir) { Dir[pattern] } : []
   end
+
+  def cpuid_instruction?(file, objdump = "objdump")
+    has_cpuid_instruction = false
+    Utils.popen_read(objdump, "--disassemble", file) do |io|
+      until io.eof?
+        has_cpuid_instruction = io.readline.include? "cpuid"
+        break if has_cpuid_instruction
+      end
+    end
+
+    has_cpuid_instruction
+  end
```

Comment:
Not sure if this is the best way to do this, but the idea here is that I don't want to wait for `objdump` to finish dumping its output (which could take a while) when it has already produced a line containing a `cpuid` instruction.

This makes `brew audit --strict openblas` run almost twice as fast compared to the naive approach of
```ruby
Utils.popen_read(objdump, "-d", file).include? "cpuid"
```

Reply:
Yeah, I think doing it this way makes sense (we only need one instance of `cpuid` to "pass" so no need to potentially waste time in my opinion. Especially because this audit could take a while if there's only one instance and it happens to be in the last file in `binary_executable_or_library_files`.

The only thing I'd ask is whether if should be `.include? "cpuid"` or whether we should use a regex or something to check that the use of `cpuid` is actually as the instruction rather than somewhere else. If strings show up in this output it's probably a good idea, but otherwise may not be a huge concern because the it seems the output is mostly going to be a defined set of instructions (and hex values)

Reply:
We, we could actually do a check like
```ruby
io.readline.split("\t")[2] == "cpuid"
```
and this works for macOS `objdump` and `llvm-objdump`, but not `binutils` `objdump`. 😞 Let me see if there's a stricter match that works for whichever one...

Reply:
I could also do something like this (macOS `objdump` is also from LLVM):
```suggestion
  def cpuid_instruction?(file, objdump = "objdump")
    objdump_is_llvm = Utils.popen_read(objdump, "--version").include? "LLVM"
    has_cpuid_instruction = false
    Utils.popen_read(objdump, "--disassemble", file) do |io|
      until io.eof?
        has_cpuid_instruction = if objdump_is_llvm
          io.readline.split("\t")[1] == "cpuid"
        else
          io.readline.include? "cpuid"
        end
        break if has_cpuid_instruction
      end
    end

    has_cpuid_instruction
  end
```

Reply:
I think I'd rather be consistent across `objdump`s than get the slightly more accurate detection in this case. Otherwise I anticipate painful debug sessions ("why does this pass on CI but not on my machine" always sucks).

If my understanding is correct based on your comment below, the `runtime_cpu_detection_allowlist` allowlist means that there's some level of manual auditing that should be done before a formula is allowed to even enter this check, so I think we should be okay. If you're super concerned, you can also add yourself to `CODEOWNERS` for the allowlist file so you can be notified and verify additions (although doing so is definitely not required).

Reply:
I would be more okay with the proposed solution if there was a warning when using non-llvm `objdump` (i.e. the one that does `io.readline.include? "cpuid"` instead of the `==`) letting the user know that the output they see may not be as accurate unless they install `llvm` to use its `objdump` (or just look at CI)